File: BDS/Java/Swizilla/src/Account.java
class Account {
	public enum Type {
		SavingsAccount, CurrentAccount
	}

	private Type accountType;
	private int accountNumber;
	private String holdersName;
	private double balance;

	public Type getAccountType() {
		return accountType;
	}

	public void setAccountType(Type accountType) {
		this.accountType = accountType;
	}

	public int getAccountNumber() {
		return accountNumber;
	}

	public void setAccountNumber(int accountNumber) {
		this.accountNumber = accountNumber;
	}

	public String getHoldersName() {
		return holdersName;
	}

	public void setHoldersName(String holdersName) {
		this.holdersName = holdersName;
	}

	public double getBalance() {
		return balance;
	}

	public void setBalance(double balance) {
		this.balance = balance;
	}

	public Account(Type accountType, int accountNumber, String holdersName, double balance) {
		this.accountType = accountType;
		this.accountNumber = accountNumber;
		this.holdersName = holdersName;
		this.balance = balance;
	}

	@Override
	public String toString() {
		return "{" + "Account Type: " + accountType + ", " + "Account Number: " + accountNumber + ", "
				+ "Holder's Name: " + holdersName + ", " + "Balance: " + balance + "}";

	}

	public double withdraw(double amount) throws Exception {
		switch (accountType) {
		case SavingsAccount:
			// SavingsAccount needs to maintain minimum balance
			if (balance - amount < 500)
				throw new Exception("Insufficient Balance Exception");
			balance -= amount;
			break;

		case CurrentAccount:
			// CurrentAccount can be 0 balance, but check amount
			if (balance - amount < 0)
				throw new Exception("Insufficient Balance Exception");
			balance -= amount;
			break;
		}
		return balance;
	}

	public double deposit(double amount) {
		switch (accountType) {
		case SavingsAccount:
		case CurrentAccount:
			// Add the amount to balance
			balance += amount;
			break;
		}
		return balance;
	}
}

File: BDS/CSharp/Swizilla/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Swizilla
{
    class Account
    {
        public enum Type { SavingsAccount, CurrentAccount }

        public Type AccountType { get; }
        public int AccountNumber { get; }
        public string HoldersName { get; }
        
        public decimal Balance { private get; set; }
        

        public Account(Type accountType, int accountNumber, string holdersName, decimal balance)
        {
            AccountType = accountType;
            AccountNumber = accountNumber;
            HoldersName = holdersName;
            Balance = balance;
        }

        public override string ToString()
        {
            return
                "{" +
                "Account Type: " + AccountType + ", " +
                "Account Number: " + AccountNumber + ", " +
                "Holder's Name: " + HoldersName + ", " +
                "Balance: " + Balance.ToString("N") +
                "}";

        }

        public decimal Withdraw(decimal amount)
        {
            switch (AccountType)
            {
                case Type.SavingsAccount:
                    // SavingsAccount needs to maintain minimum balance
                    if (Balance - amount < 500)
                        throw new Exception("Insufficient Balance Exception");
                    Balance -= amount;
                    break;

                case Type.CurrentAccount:
                    // CurrentAccount can be 0 balance, but check amount
                    if (Balance - amount < 0)
                        throw new Exception("Insufficient Balance Exception");
                    Balance -= amount;
                    break;
            }
            return Balance;
        }

        public decimal Deposit(decimal amount)
        {
            switch (AccountType)
            {
                case Type.SavingsAccount:
                case Type.CurrentAccount:
                    // Add the amount to balance
                    Balance += amount;
                    break;
            }
            return Balance;
        }

    }
}

